test_scripts/MobileProjection/Defects/3142_01_Absence_of_misplaced_OnVideoDataStreaming.lua

```diff
@@ -0,0 +1,84 @@
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/3142
+--
+-- Steps:
+-- 1. Set StopStreamingTimeout = 3000 and VideoDataStoppedTimeout = 3000 in SDL .INI file
+-- 2. Start SDL, HMI, connect Mobile device
+-- 3. Register PROJECTION (App_1) and NAVIGATION (App_2) applications
+-- 4. Activate App_1 and start Video streaming
+-- 5. Deactivate App_1 (streaming still continue since app has STREAMABLE state)
+-- 6. Activate App_2 and start Video streaming
+-- 7. App_1 and App_2 continue streaming data withing 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - switch streaming between apps and provides HMI with streaming data from App_2
+--   - not unregister App_1 since timeout is not yet expired
+-- 8. App_1 stops streaming within 'StopStreamingTimeout' timeout
+-- SDL does:
+--   - not send Navi.OnVideoDataStreaming(false) notification to HMI since App_2 still continue streaming
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("test_scripts/MobileProjection/Defects/common")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Apps Configuration ]]
+common.app.getParams(1).appHMIType = { "PROJECTION" }
+common.app.getParams(2).appHMIType = { "NAVIGATION" }
+
+--[[ Local Functions ]]
+local function stopStreaming(pAppId, pServiceId)
+  common.hmi.getConnection():ExpectRequest("Navigation.StopStream", { appID = common.app.getHMIId(pAppId)})
+  :Do(function(_, data) common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { }) end)
+  common.mobile.getSession(pAppId):ExpectEndService(pServiceId)
+  :Do(function() common.mobile.getSession(pAppId):SendEndServiceAck(pServiceId) end)
+end
+
+local function startStreaming(pAppId, pServiceId)
+  common.mobile.getSession(pAppId):StartService(pServiceId)
+  :Do(function()
+      common.mobile.getSession(pAppId):StartStreaming(pServiceId, common.streamFiles[pAppId], 160*1024)
+      common.log("App " .. pAppId .." starts streaming ...")
+      common.streamingStatus[pAppId] = true
+    end)
+  common.hmi.getConnection():ExpectRequest("Navigation.StartStream", { appID = common.app.getHMIId(pAppId) })
+  :Do(function(_, data) common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", { }) end)
+end
+
+local function activateApp2()
+  local cid = common.hmi.getConnection():SendRequest("SDL.ActivateApp", { appID = common.app.getHMIId(2) })
+  common.hmi.getConnection():ExpectResponse(cid)
+  common.mobile.getSession(2):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "FULL", videoStreamingState = "STREAMABLE" })
+  :Do(function()
+      startStreaming(2, 11)
+    end)
+  stopStreaming(1, 11)
+  common.hmi.getConnection():ExpectNotification("Navigation.OnVideoDataStreaming",
+    { available = false }, { available = true }):Times(2)
```

Comment:
I think that in this function implementation HMI should not receive the `OnVideoDataStreaming(false)` because the Navi app starts stream right after activation App2 and SDL should stop a transferring of stream data from App1 also after activation App2, but SDL has timeout `VideoDataStoppedTimeout=3000` so SDL will send `OnVideoDataStreaming(false)` only in 3 sec after  last received stream data from mobile.

Reply:
@GetmanetsIrina The reasons of `false` message is that Mobile connection could be slow however HMI connection is always fast, hence HMI should get `false` status as soon as possible (in our case almost at the same time as user activates 2nd app) in order to stop streaming internally.